src/components/TodosList.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { TodosFilter } from '../types/TodosFilter';
+
+type Props = {
+  currentTodos: Todo[];
+  filter: TodosFilter;
+  setCurrentTodos: (updatedTodos: Todo[]) => void;
+  updatingTodo: boolean;
+  newTodo: string;
+};
+
+export const TodoList: React.FC<Props> = ({
+  currentTodos,
+  filter,
+  setCurrentTodos,
+  updatingTodo,
+  newTodo,
+}) => {
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+
+  const filteredTodos = () => {
+    switch (filter) {
+      case TodosFilter.all:
+        return currentTodos;
+      case TodosFilter.completed:
+        return currentTodos.filter((todo) => todo.completed);
+      case TodosFilter.active:
+        return currentTodos.filter((todo) => !todo.completed);
+      default:
+        return currentTodos;
+    }
+  };
+
+  const handleDeleteCurrentTodo = (todoId: number) => {
+    setDeletingTodoId(todoId);
+
+    setTimeout(() => {
+      const updatedTodos = currentTodos.filter((todo) => todo.id !== todoId);
+
+      setCurrentTodos(updatedTodos);
+      setDeletingTodoId(null);
+    }, 300);
```

Comment:
You need to send a request

<img width="404" alt="image" src="https://github.com/mate-academy/react_todo-app-add-and-delete/assets/49764881/9d47bbb3-a44d-47cd-a534-b3c191011586">